future-apps/payroll/contracts/Payroll.sol

```diff
@@ -137,192 +139,145 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
+     * @param _denominationSalary Employee's new salary, per second in denomination token.
      */
-    function setEmployeeSalary(
-        uint256 _employeeId,
-        uint256 _denominationSalary
-    )
+    function setEmployeeSalary(uint256 _employeeId, uint256 _denominationSalary)
         external
-        employeeActive(_employeeId)
         authP(SET_EMPLOYEE_SALARY_ROLE, arr(_employeeId, _denominationSalary))
+        employeeActive(_employeeId)
     {
-        uint64 timestamp = getTimestamp64();
-
         // Add owed salary to employee's accrued value
-        uint256 owed = _getOwedSalary(_employeeId, timestamp);
+        uint256 owed = _getOwedSalary(_employeeId);
         _addAccruedValue(_employeeId, owed);
 
         // Update employee to track the new salary and payment date
         Employee storage employee = employees[_employeeId];
-        employee.lastPayroll = timestamp;
+        employee.lastPayroll = getTimestamp64();
         employee.denominationTokenSalary = _denominationSalary;
 
         emit SetEmployeeSalary(_employeeId, _denominationSalary);
     }
 
     /**
-     * @notice Terminate employee #`_employeeId`
-     * @param _employeeId Employee's identifier
+     * @notice Terminates employee #`_employeeId`.
+     * @param _employeeId Employee's identifier.
      */
-    function terminateEmployeeNow(
-        uint256 _employeeId
-    )
+    function terminateEmployeeNow(uint256 _employeeId)
         external
-        employeeActive(_employeeId)
         authP(TERMINATE_EMPLOYEE_ROLE, arr(_employeeId))
+        employeeActive(_employeeId)
     {
         _terminateEmployee(_employeeId, getTimestamp64());
     }
 
     /**
-     * @notice Terminate employee #`_employeeId` on `@formatDate(_endDate)`
-     * @param _employeeId Employee's identifier
-     * @param _endDate Termination date
+     * @notice Terminates employee #`_employeeId` on `@formatDate(_endDate)`.
+     * @param _employeeId Employee's identifier.
+     * @param _endDate Termination timestamp in seconds.
      */
-    function terminateEmployee(
-        uint256 _employeeId,
-        uint64 _endDate
-    )
+    function terminateEmployee(uint256 _employeeId, uint64 _endDate)
         external
-        employeeActive(_employeeId)
         authP(TERMINATE_EMPLOYEE_ROLE, arr(_employeeId))
+        employeeActive(_employeeId)
     {
         _terminateEmployee(_employeeId, _endDate);
     }
 
     /**
-     * @notice Adds `_amount` to accrued value for employee #`_employeeId`
-     * @param _employeeId Employee's identifier
-     * @param _amount Amount to add
+     * @notice Adds `_amount` to accrued value for employee #`_employeeId`.
+     * @param _employeeId Employee's identifier.
+     * @param _amount Amount be added to the employee's accrued value.
      */
-    function addAccruedValue(
-        uint256 _employeeId,
-        uint256 _amount
-    )
+    function addAccruedValue(uint256 _employeeId, uint256 _amount)
```

Comment:
Let's follow this conversation [here](https://github.com/aragon/aragon-apps/pull/744/files?file-filters%5B%5D=.sol&hide-deleted-files=true#r271881379) just to avoid multiple threads with the same topic :)